employee-rostering-backend/src/test/java/org/optaweb/employeerostering/skill/SkillRestControllerTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.skill;
+
+import java.util.List;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.optaweb.employeerostering.domain.skill.Skill;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class SkillRestControllerTest {
+
+    @Autowired
+    private TestRestTemplate skillRestTemplate;
+
+    private String skillPathURI = "http://localhost:8080/rest/tenant/{tenantId}/skill/";
+
+    private ResponseEntity<List<Skill>> getSkills(Integer tenantId) {
+        return skillRestTemplate.exchange(skillPathURI, HttpMethod.GET, null, new ParameterizedTypeReference
+                <List<Skill>>() {}, tenantId);
```

Comment:
Code style: The point between class name and its type parameters is not the best place for a line break. Here, the whole expression creating an instance of the anonymous class functions as a single argument. Except for the type param, it's pretty atomic so it's a bit confusing that it's interrupted in the middle and continues on the next line.

Space between parameters is a better place for a line break.

Another option is to create the instance and store it in a variable as a separate statement:
```java
ParameterizedTypeReference<List<Spot>> typeReference = new ParameterizedTypeReference<List<Spot>>() {};
return spotRestTemplate.exchange(spotPathURI, HttpMethod.GET, null, typeReference, tenantId);
```